client/service/test/src/client/mod.rs

```diff
@@ -1483,34 +1500,35 @@ fn respects_block_rules() {
 		// Now going to the fork
 		block_on(client.import_as_final(BlockOrigin::Own, block_ok)).unwrap();
 
-		// And check good fork
-		let mut block_ok =
-			client.new_block_at(&BlockId::Number(1), Default::default(), false).unwrap();
+		// And check good fork (build B[2])
+		let mut block_ok = client.new_block_at(block_ok_1_hash, Default::default(), false).unwrap();
 		block_ok.push_storage_change(vec![0], Some(vec![2])).unwrap();
 		let block_ok = block_ok.build().unwrap().block;
+		assert_eq!(*block_ok.header().number(), 2);
 
 		let params = BlockCheckParams {
 			hash: block_ok.hash(),
-			number: 1,
+			number: 2,
 			parent_hash: *block_ok.header().parent_hash(),
 			allow_missing_state: false,
 			allow_missing_parent: false,
 			import_existing: false,
 		};
 		if record_only {
-			fork_rules.push((1, block_ok.hash()));
+			fork_rules.push((2, block_ok.hash()));
```

Comment:
Checking fork at block height 2 is more readable and less confusing, this is also inline with actual block number, and also with params passed to `check_block`.